primitives/xcm/src/fee_handlers.rs

```diff
@@ -82,7 +82,7 @@ impl<
 				// This involves the same db read per block, mitigating any attack based on
 				// non-supported assets
 				if !AssetIdInfoGetter::payment_is_supported(asset_type.clone()) {
-					return Err(XcmError::TooExpensive);
+					return Err(XcmError::AssetNotFound);
```

Comment:
Adding a new error named `AssetNotSupported` is preferable.

Reply:
This part will be remove by this PR anyway, it's not ready for review